applications/IgaApplication/custom_modelers/iga_modeler_sbm.cpp

```
@@ -0,0 +1,601 @@
+    
+    const std::string surrogate_sub_model_part_name = is_inner ? "surrogate_inner" : "surrogate_outer";
+    
+    ModelPart& surrogate_sub_model_part = rModelPart.GetParentModelPart().HasSubModelPart(surrogate_sub_model_part_name)
+            ? rModelPart.GetParentModelPart().GetSubModelPart(surrogate_sub_model_part_name)
+            : KRATOS_ERROR << "::[CreateQuadraturePointGeometriesSbm]::: Sbm case -> surrogate_sub_model_part has not been defined before."
+                            << "Maybe you are not calling the nurbs_modeler_sbm" << std::endl;
+
+    if (is_inner) { // INNER
+        for (auto &i_cond : skin_sub_model_part_in.Conditions()) {
+            points.push_back(PointTypePointer(new PointType(i_cond.Id(), i_cond.GetGeometry().Center().X(), i_cond.GetGeometry().Center().Y(), i_cond.GetGeometry().Center().Z())));
+        }
+    } 
+    else { // OUTER
+        for (auto &i_cond : skin_sub_model_part_out.Conditions()) {
+            points.push_back(PointTypePointer(new PointType(i_cond.Id(), i_cond.GetGeometry().Center().X(), i_cond.GetGeometry().Center().Y(), i_cond.GetGeometry().Center().Z())));
+        }
+    }
+    
+    // Get the mesh sizes from the surrogate model part
+    const Vector knot_span_sizes = surrogate_sub_model_part.GetParentModelPart().GetValue(KNOT_SPAN_SIZES);
+    // Get the parameter space corners from the surrogate model part
+    const std::vector<Vector> parameter_space_corners = surrogate_sub_model_part.GetParentModelPart().GetValue(PARAMETER_SPACE_CORNERS);
+
+    double knot_span_reference_size = knot_span_sizes[0];
+    if (knot_span_sizes[1] > knot_span_reference_size) {knot_span_reference_size = knot_span_sizes[1];}
+    if (knot_span_sizes.size() > 2) {if (knot_span_sizes[2] > knot_span_reference_size) {knot_span_reference_size = knot_span_sizes[2];}}
+
+    // 2D case
+    const double search_radius = sqrt(2)*(knot_span_reference_size); 
+    // 3D case //TODO:
```

If 3D implementation is pending, I'd throw an error. You can check the dimension from the `ProcessInfo` `DOMAIN_SIZE` value.